### ISSUE TITLE
TypeError: decode() argument 2 must be str, not None

### ISSUE THREAD

**Issue by espdev** [mikeboers/PyAV#265]
Hi!

I try to open some video file and I get an error:

```python
>>> import av
>>> av.open('test_video_38.mp4')
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "av\container\core.pyx", line 249, in av.container.core.open
  File "av\container\input.pyx", line 50, in av.container.input.InputContainer.__cinit__
  File "av\stream.pyx", line 45, in av.stream.wrap_stream
  File "av\stream.pyx", line 64, in av.stream.Stream._init
  File "av\utils.pyx", line 137, in av.utils.avdict_to_dict
  File "av\utils.pyx", line 120, in av.utils._decode
TypeError: decode() argument 2 must be str, not None
```

The video file is available on link:
https://yadi.sk/i/bGzjHbvC3LtBf7

Also I get media info about the file by [MediaInfo](https://mediaarea.net/ru/MediaInfo) library:
https://paste.fedoraproject.org/paste/i5-mJObDMPGkn3bo~Kht2w

It seems that the file is OK.
Why PyAV cannot open the file?

**Comment by jmoraleda** [comment]
I experience the same issue. I tryed applying the patch suggested by @mkfmnn (initializing ContainerProxy after metadata) but I still get the same error.

**Comment by jmoraleda** [comment]
I am experiencing the same issue. I tried applying the patch suggested by @mkfmnn (initializing ContainerProxy after metadata in core.pyx) but it did not fix the problem for me. I am running the latest code from github.

**Comment by jmoraleda** [comment]
I decided to do a little bit of debugging by checking out earlier commits, but the results are strange:

I went back through several commits to aeca78d (from 5/6/2017) which still failed 
then I jumped back to 41ab2a94 which worked...

BUT then I went back to aeca78d and this time it WORKED ... and then I went back to the head and it ALSO WORKED.

My guess is that there is some race condition in the build system, perhaps in initialization of the c libraries, that gets triggered or not depending on your luck when building the libraries, but I would have to do more testing to validate this hunch. Hopefully this report helps though.

**Comment by mkfmnn** [comment]
@jmoraleda, is it possible you're picking up build artifacts from a previous build, or perhaps from a different location on your filesystem? Can you retry my patch after re-running `make clean-all`? I'm pretty sure that this error (which is Python 3-specific) should be completely deterministic and not based on library initialization order.

**Comment by jmoraleda** [comment]
@mkfmnn Thank you for your quick reply! 
Yes it is possible that I was picking up old artifacts as I have been using the version from github for a while and I updated today after a few weeks, but I did not do a make clean.
Yes. I am using python3 (I replace all python calls  in Makefile with calls to python3. Is there a better way?)

After I run `make clean-all `and then `make` on the head of master the bug did not appear, so I did not get to try if the patch would make any difference.
